packages/cardhost/app/lib/card-model-for-browser.ts

```diff
@@ -265,6 +262,32 @@ export default class CardModelForBrowser implements CardModel {
     return s;
   }
 
+  @restartableTask async rerenderData(
+    data: Record<string, any>
+  ): Promise<void> {
+    this.state.rawData = merge({}, this.state.rawData, {
+      attributes: data,
+    });
+    let newSchemaInstance = await this.createSchemaInstance();
```

Comment:
I'm confused why we need to instantiate a new instance every time we rerender. Why is this necessary?

Reply:
we are instantiating a new instance because the `this._schemaInstance` is a tracked property. We want to have the rerenders of the template present consistent data for all the fields, such that we don't see a field consumed by a computed being rerendered at a different time than the computed field that consumes it. all the field updates need to atomic which includes any computeds (anyc or sync) that consume the fields. In order to do this we create a "working" schema instance that we use for the intermediate states, and when the schema instance has settled, we then set it to the tracked tracked property for a consistent rendering of the template.